pkg/discovery/discovery.go

```diff
@@ -31,6 +32,9 @@ import (
 // DefaultWeight is the default weight for an instance.
 const DefaultWeight = 10
 
+// ErrInstanceNotFound means instance not found
+var ErrInstanceNotFound = errors.New("instance not found")
```

Comment:
Maybe it is better that add this Error definition in kerrors.go.  eg 
`ErrInstanceNotFound     = ErrServiceDiscovery.WithCause(errors.New("instance not found"))`

Reply:
done.